src/test/java/uk/gov/ons/census/casesvc/service/ReceiptProcessorTest.java

```diff
@@ -1,118 +1,98 @@
+import static uk.gov.ons.census.casesvc.testutil.DataUtils.getTestResponseManagementEvent;
 
 import java.time.OffsetDateTime;
-import java.util.HashMap;
-import java.util.Map;
 import java.util.Optional;
 import java.util.UUID;
-import org.jeasy.random.EasyRandom;
 import org.junit.Test;
-import uk.gov.ons.census.casesvc.model.dto.PayloadDTO;
-import uk.gov.ons.census.casesvc.model.dto.Receipt;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+import uk.gov.ons.census.casesvc.logging.EventLogger;
+import uk.gov.ons.census.casesvc.model.dto.ReceiptDTO;
+import uk.gov.ons.census.casesvc.model.dto.ResponseManagementEvent;
 import uk.gov.ons.census.casesvc.model.entity.Case;
 import uk.gov.ons.census.casesvc.model.entity.EventType;
 import uk.gov.ons.census.casesvc.model.entity.UacQidLink;
 import uk.gov.ons.census.casesvc.model.repository.CaseRepository;
 import uk.gov.ons.census.casesvc.model.repository.UacQidLinkRepository;
 
+@RunWith(MockitoJUnitRunner.class)
 public class ReceiptProcessorTest {
   private static final UUID TEST_CASE_ID = UUID.randomUUID();
-  private static final String TEST_QID = "test_qid";
-  private static final String TEST_UAC = "test_uac";
+  private static final String TEST_QUESTIONNAIRE_ID = "123";
+
+  @Mock private CaseProcessor caseProcessor;
+
+  @Mock private UacQidLinkRepository uacQidLinkRepository;
+
+  @Mock private CaseRepository caseRepository;
+
+  @Mock private UacProcessor uacProcessor;
+
+  @Mock private EventLogger eventLogger;
+
+  @InjectMocks
+  ReceiptProcessor underTest;
 
   @Test
   public void testGoodReceipt() {
-    CaseRepository caseRepository = mock(CaseRepository.class);
-    UacQidLinkRepository uacQidLinkRepository = mock(UacQidLinkRepository.class);
+    ResponseManagementEvent managementEvent = getTestResponseManagementEvent();
+    ReceiptDTO expectedReceipt = managementEvent.getPayload().getReceipt();
 
     // Given
-    UacQidLink expectedUacQidLink = getUacQidLink();
     Case expectedCase = getRandomCase();
+    UacQidLink expectedUacQidLink = expectedCase.getUacQidLinks().get(0);
     expectedUacQidLink.setCaze(expectedCase);
 
-    when(uacQidLinkRepository.findByQid(TEST_QID)).thenReturn(Optional.of(expectedUacQidLink));
-
-    UacProcessor uacProcessor = mock(UacProcessor.class);
-    CaseProcessor caseProcessor = mock(CaseProcessor.class);
-    PayloadDTO payloadDTO = new PayloadDTO();
+    managementEvent.getPayload().getReceipt().setResponseDateTime(OffsetDateTime.now());
 
-    when(uacProcessor.emitUacUpdatedEvent(any(UacQidLink.class), any(Case.class), anyBoolean()))
-        .thenReturn(payloadDTO);
-    when(caseProcessor.emitCaseCreatedEvent(any(Case.class))).thenReturn(new PayloadDTO());
-
-    Map<String, String> headers = createTestHeaders();
+    when(uacQidLinkRepository.findByQid(anyString())).thenReturn(Optional.of(expectedUacQidLink));
```

Comment:
This should be similar to before where we expect an explicit value, this tests as we go and is nicely functional. Something like below uses the random qid we generate. 
```
when(uacQidLinkRepository.findByQid(managementEvent.getPayload().getReceipt().getQuestionnaireId()))
     .thenReturn(Optional.of(expectedUacQidLink));